RISTNet.cpp

```diff
@@ -153,12 +159,18 @@ int RISTNetReceiver::clientDisconnect(void *pArg, rist_peer *pPeer) {
         return 0;
     }
 
-    if (lWeakSelf->mClientList.find(pPeer) == lWeakSelf->mClientList.end()) {
+	auto netObj = lWeakSelf->mClientList.find(pPeer);
+	if (netObj == lWeakSelf->mClientList.end()) {
         LOGGER(true, LOGG_ERROR, "RISTNetReceiver::clientDisconnect unknown peer")
         return 0;
-    } else {
-        lWeakSelf->mClientList.erase(lWeakSelf->mClientList.find(pPeer)->first);
+	}
+
+    if (lWeakSelf->clientDisconnectedCallback) {
+        lWeakSelf->clientDisconnectedCallback(netObj->second, *pPeer);
+		lWeakSelf->clientDisconnectedCallback = nullptr;
     }
+
+	lWeakSelf->mClientList.erase(netObj->first);
```

Comment:
Use pPeer instead of netObj->first, even if they are the same reference. Since you do the map lookup with the pPeer variable it makes the code easier to understand if you erase with the same variable name.

Reply:
Nice!